invenio/modules/relationships/engines/sqlalchemy.py

```diff
@@ -0,0 +1,117 @@
+from uuid import uuid4
+
+from sqlalchemy_utils import UUIDType
+
+from invenio.ext.sqlalchemy import db, utils
+
+
+class SQLAlchemyNode(object):
```

Comment:
As every implementation of `Node` and every implementation of `Relationship` will have the same interface, it might be worth to create "abstract" superclasses for them.
Such solution is implemented for example for `StorageEngines` in `JSONAlchemy`:

https://github.com/inveniosoftware/invenio/blob/master/invenio/modules/jsonalchemy/jsonext/engines/sqlalchemy.py
https://github.com/inveniosoftware/invenio/blob/master/invenio/modules/jsonalchemy/storage.py

After doing so, a new developer who will work on another engines, will __surely__ implement API methods. If they are well documented, his methods will give expected output.